Shared/Extensions/URLExtensions.swift

```diff
@@ -192,6 +192,22 @@ extension URL {
             return urlString
         }
     }
+    
+    public var typedDisplayString: String {
+        var urlString = self.absoluteString
+
+        if urlString.hasPrefix("http://") {
+            urlString = String(urlString[urlString.index(urlString.startIndex, offsetBy: 7)...])
+        } else if urlString.hasPrefix("https://") {
+            urlString = String(urlString[urlString.index(urlString.startIndex, offsetBy: 8)...])
+        }
+        
+        if urlString.hasSuffix("/") {
+            urlString.removeLast()
+        }
+        
+        return urlString
+    }
```

Comment:
1. Missing documentation as to what this does
2. Missing tests around it

But why do we need this anyways?

Reply:
The problem is according to visitType (checking if the visit type is .typed url) we need to record all the navigation that is made before we add history item. The reason is brave-core side only wants us to sync the typed urls. This is what we gather from browser and we are forced to mimic this behaviour. 

Inside the `func navigateInTab(tab: Tab, to navigation: WKNavigation? = nil)` is where the history registry is made before and I was careful not to make any changes to how url is retrieved and where it is being added.

`typedNavigation` is just a dictionary where the display url is the key and visitType is value. According to this url is from bookmarks, typed, linked etc.. the url key might be different because of scheme existence or the "/" suffix. So we have to strip out these from the url string, I wanted to use `absoluteDisplayString` however it was behaving differently when the url is not secure.

Test and documentation is added fix at https://github.com/brave/brave-ios/pull/3703/commits/e5d686c76f61778b93219063462a2703f7ab2bf4

Reply:
`url.schemelessAbsoluteString` might help?

Reply:
Changed the implementation using `schemelessAbsoluteString`
Fix at https://github.com/brave/brave-ios/pull/3703/commits/382666ee2b53cd730e44b957254c81880e89757b